clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -240,9 +237,8 @@ protected boolean flush(ByteBuffer buf) throws IOException {
     */
     @Override
     public void handshake() throws IOException {
-        // Reset state to support renegotiation. This can be removed if renegotiation support is removed.
-        if (state == State.READY)
-            state = State.HANDSHAKE;
+        if (state != State.HANDSHAKE)
+            throw renegotiationException();
```

Comment:
We can also get here if handshake has already failed (state == `State.HANDSHAKE_FAILED`) and there are still bytes to be flushed in `netWriteBuffer`.